addons/hr_attendance/models/hr_attendance.py

```diff
@@ -335,8 +335,20 @@ def _update_overtime(self, employee_attendance_dates=None):
                 # Overtime is not counted if any shift is not closed or if there are no attendances for that day,
                 # this could happen when deleting attendances.
                 if not unfinished_shifts and attendances:
+                    # The employee is working flexible hours
+                    if emp.resource_id._is_flexible():
```

Comment:
why not also having those methods inside hr.employee model since the calendar is also defined on the employee. Actually, I think we use the resource_calendar_id field from hr.employee instead of the one in calendar_id when we got `hr.employee` records. 🙂 

Actually, I think it could be a good idea to have 2 computed non-stored fields instead of just those 2 methods.

Something like:

```python
    is_flexible = fields.Boolean(compute='_compute_is_flexible', search='_search_is_flexible')
    is_fully_flexible = fields.Boolean(compute='_compute_is_flexible', search='_search_is_fully_flexible')

    @api.depends('resource_calendar_id')
    def _compute_is_flexible(self):
        for employee in self:
            employee.is_fully_flexible = not self.resource_calendar_id
            employee.is_flexible = employee.is_fully_flexible or self.resource_calendar_id.flexible_hours

    def _search_is_fully_flexible(self, op, value):
        return ['!', ('resource_calendar_id', op, value)]

    def _search_is_flexible(self, operator, value):
        if operator not in ('=', '!=') or isinstance(value, bool):
            raise ValidationError(_("The operator should be '=' or '!=' and the value should be a boolean value."))
        # '!=' True or '=' False
        if (operator == '=') ^ value:
            return [('resource_calendar_id', '!=', False), ('resource_calendar_id.flexible_hours', '=', False)]
        # '=' True or '!=' False
        else:
            return ['|', ('resource_calendar_id', '=', False), ('resource_calendar_id.flexible_hours', '=', True)]
```

By doing that, we could use `is_flexible` and/or `is_fully_flexible` fields inside domains since a search method is defined and we could use those fields in our conditions instead of calling the methods. 🙂

Reply:
Thank you very much for the suggestion!